lib/controllers/register.js

```diff
@@ -125,159 +125,174 @@ module.exports = function (req, res) {
   var postRegistrationHandler = config.postRegistrationHandler;
   var view = config.web.register.view;
 
-  // Handle incoming POST requests from an API-like clients (something like
-  // Angular / React / REST).
-  if (req.method === 'POST' && accepts === 'json') {
-    applyDefaultAccountFields(config, req);
-    helpers.validateAccount(req.body, config, function (errors) {
-      if (errors) {
-        return res.status(400).json({ error: errors[0].message });
+  helpers.handleAcceptRequest(req, {
+    // Handle incoming POST requests from an API-like clients (something like
```

Comment:
I would also remove the `POST` word here, even though it currently only handles `POST` requests.